test/dlc_tests/conftest.py

```diff
@@ -222,10 +235,105 @@ def pytest_runtest_setup(item):
             pytest.skip("Skipping non-canary tests")
 
 
+def pytest_collection_modifyitems(session, config, items):
+    if config.getoption("--generate-coverage-doc"):
+        test_cov = {}
+        for item in items:
+            # Define additional csv options
+            function_name = item.name.split("[")[0]
+            function_key = f"{item.fspath}::{function_name}"
+            str_fspath = str(item.fspath)
+            str_keywords = str(item.keywords)
+
+            # Based on keywords and filepaths, assign values
+            scope = infer_field_value("all", ("mxnet", "tensorflow", "pytorch"), str_fspath)
+            train_inf = infer_field_value("both", ("training", "inference"), str_fspath, str_keywords)
+            integration = infer_field_value("general integration", ("_dgl_", "_smdebug_", "_gluonnlp_"), str_keywords)
+            model = infer_field_value("N/A", ("mnist", "densenet"), str_keywords)
```

Comment:
These are basically to guess what the integration is based on the test name and other metadata in the case where someone did not use the integration marker.